src/program/latex.rs

```diff
@@ -0,0 +1,118 @@
+use crate::Program;
+use generation::{LatexFactory, diagram::DiagramFactory};
+
+mod generation;
+
+pub struct Request {
+    document: Document,
+    settings: Option<Settings>,
+}
+
+impl Default for Request {
+    fn default() -> Self {
+        Self { document: Document::Diagram, settings:  None}
+    }
+}
+
+pub enum Document {
+    Diagram,
+    None,
```

Comment:
Elsewhere seems to indicate that `None` causes errors -- is there a case where it is valid? If not, why include it?